membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -563,9 +595,45 @@ def test_analysis_wrapping_coordinates(self, x_bin, y_bin, box_dim, dummy_array,
                                y_range=y_range).run()
         avg_surface = mc.results.average_z_surface
         # assert if default values of wrapped coords in z_surface returns correctly
-        assert_almost_equal(avg_surface, expected_surface)
+        assert_allclose(avg_surface, expected_surface)
 
     def test_test_analysis_no_wrapping(self, universe):
         regex = (r"`wrap == False` may result in inaccurate calculation")
         with pytest.warns(UserWarning, match=regex):
             MembraneCurvature(universe, wrap=False)
+
+    # test curvature with interpolated surface
+    @pytest.mark.parametrize('dim_x, dim_y, x_bins, y_bins, dummy_array, expected_interp_surf', [
+        # array 3x3 with all 150 and one nan
+        (300, 300, 3, 3, np.array([[0., 0.,   150.], [100., 0.,   150.],   [200., 0.,   150.],
+                                   [0., 100., 150.], [100., 100., np.nan], [200., 100., 150.],
+                                   [0., 200., 150.], [100., 200., 150.],   [200., 200., 150.]]),
+         np.full((1, 3, 3), 150.)),
+        # array 3x3 with all 150 and one nan
+        (300, 300, 3, 3, np.array([[0., 0.,   150.], [100., 0.,   150.],   [200., 0.,   150.],
+                                   [0., 100., 150.], [100., 100., np.nan], [200., 100., 150.],
+                                   [0., 200., np.nan], [100., 200., 150.],   [200., 200., 150.]]),
+         np.full((1, 3, 3), 150.)),
+        # array 3x4 with all 150 and three nans
+        (300, 400, 3, 4, np.array([[0., 0.,   150.], [100., 0.,   150.],   [200., 0.,   150.],
+                                   [0., 100., 150.], [100., 100., np.nan], [200., 100., 150.],
+                                   [0., 200., 150],  [100., 200., np.nan], [200., 200., np.nan],
+                                   [0., 300., 150.], [100., 300., 150.],   [200., 300., 150.]]),
+         np.full((1, 3, 4), 150.)),
+        # array 4x4 with all 120 and many nans
+        (400, 400, 4, 4, np.array([[0., 0., np.nan], [100., 0., 120.], [200., 0., 120.], [300., 0., np.nan],
+                                   [0., 100., 120.], [100., 100., np.nan], [200., 100., 120.], [300., 100., 120.],
+                                   [0., 200., 120], [100., 200., np.nan], [200., 200., np.nan], [300., 200., 120.],
+                                   [0., 300., np.nan], [100., 300., 120.], [200., 300., 120.], [300., 300., np.nan]]),
+         np.full((1, 4, 4), 120.))
+    ])
```

Comment:
Here as well, tests where the expected values aren't all one number are needed to really check assumptions such as:

* is it interpolating along the axis I think it is
* is it interpolating in a linear way (vs. some random other polynomial)
* is it even interpolating, or just taking the most common number?
* are we sure it's interpolating independently for every frame, or is it interpolating across different frames?

etc.

I am, incidentally, somewhat surprised that coordinates are getting wrapped with `np.nan` values in them. @richardjgowers should this be happening...?